src/main/java/gyro/google/compute/AbstractInstanceGroupManagerResource.java

```diff
@@ -0,0 +1,446 @@
+                this,
+                "instance-template",
+                "Either 'instance-template' or 'version' is required!"));
+            errors.add(new ValidationError(
+                this,
+                "version",
+                "Either 'instance-template' or 'version' is required!"));
+        }
+        return errors;
+    }
+
+    @Override
+    public void copyFrom(InstanceGroupManager model) {
+        setBaseInstanceName(model.getBaseInstanceName());
+        setName(model.getName());
+        setTargetSize(model.getTargetSize());
+
+        List<ComputeInstanceGroupManagerAutoHealingPolicy> diffableAutoHealingPolicies = null;
+        List<InstanceGroupManagerAutoHealingPolicy> autoHealingPolicies = model.getAutoHealingPolicies();
+
+        if (autoHealingPolicies != null && !autoHealingPolicies.isEmpty()) {
+            diffableAutoHealingPolicies = autoHealingPolicies
+                .stream()
+                .map(e -> {
+                    ComputeInstanceGroupManagerAutoHealingPolicy computeInstanceGroupManagerAutoHealingPolicy = newSubresource(
+                        ComputeInstanceGroupManagerAutoHealingPolicy.class);
+                    computeInstanceGroupManagerAutoHealingPolicy.copyFrom(e);
+                    return computeInstanceGroupManagerAutoHealingPolicy;
+                })
+                .collect(Collectors.toList());
+        }
+        setAutoHealingPolicy(diffableAutoHealingPolicies);
+        setDescription(getDescription());
+        setFingerprint(model.getFingerprint());
+        setInstanceTemplate(Optional.ofNullable(model.getInstanceTemplate())
+            .map(e -> findById(InstanceTemplateResource.class, e))
+            .orElse(null));
+
+        List<ComputeNamedPort> diffableNamedPorts = null;
+        List<NamedPort> namedPorts = model.getNamedPorts();
+
+        if (namedPorts != null && !namedPorts.isEmpty()) {
+            diffableNamedPorts = namedPorts
+                .stream()
+                .map(e -> {
+                    ComputeNamedPort computeNamedPort = newSubresource(ComputeNamedPort.class);
+                    computeNamedPort.copyFrom(e);
+                    return computeNamedPort;
+                })
+                .collect(Collectors.toList());
+        }
+        setNamedPort(diffableNamedPorts);
+
+        // TODO: https://github.com/perfectsense/gyro-google-provider/issues/79
+        //        List<AbstractTargetPoolResource> diffableTargetPools = null;
+        //        List<String> targetPools = model.getTargetPools();
+        //
+        //        if (targetPools != null && !targetPools.isEmpty()) {
+        //            diffableTargetPools = targetPools
+        //                .stream()
+        //                .map(e -> findById(AbstractTargetPoolResource.class, e))
+        //                .collect(Collectors.toList());
+        //        }
+        //        setTargetPools(diffableTargetPools);
```

Comment:
Should be removed and be done as a separate PR when target pool is available

Reply:
Yeah, it should be handled in a separate PR, but better to leave the comment as it will be helpful when implementing it.